packages/discord.js/src/structures/ClientApplication.js

```diff
@@ -68,6 +68,56 @@ class ClientApplication extends Application {
       this.installParams ??= null;
     }
 
+    /**
+     * OAuth2 installation paremeters.
+     * @typedef {Object} IntegrationTypesConfigurationParameters
+     * @property {OAuth2Scopes[]} scopes Scopes that will be set upon adding this application
+     * @property {Readonly<PermissionsBitField>} permissions Permissions that will be requested for the integrated role
+     */
+
+    /**
+     * The application's supported installation context data.
+     * @typedef {Object} IntegrationTypesConfigurationContext
+     * @property {?IntegrationTypesConfigurationParameters} oauth2InstallParams
+     * Scopes and permissions regarding the installation context
+     */
+
+    /**
+     * The application's supported installation context data.
+     * @typedef {Object} IntegrationTypesConfiguration
+     * @property {IntegrationTypesConfigurationContext} [0] Scopes and permissions
+     * regarding the guild-installation context
+     * @property {IntegrationTypesConfigurationContext} [1] Scopes and permissions
+     * regarding the user-installation context
+     */
+
+    if ('integration_types_config' in data) {
+      /**
+       * Default scopes and permissions for each supported installation context.
+       * The keys are stringified variants of {@link ApplicationIntegrationType}.
+       * @type {?IntegrationTypesConfiguration}
+       */
+      this.integrationTypesConfig = Object.fromEntries(
+        Object.entries(data.integration_types_config).map(([key, config]) => {
+          let oauth2InstallParams = null;
+          if ('oauth2_install_params' in config) {
```

Comment:
According to the spec, this might be nullable. Let's just use this instead to be safe

https://github.com/discord/discord-api-spec/blob/243e3da80741492beb318a5815b99c24d719af54/specs/openapi.json#L12489-L12498

```suggestion
          if (config.oauth2_install_params) {
```